hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/solr/SolrUtil.java

```diff
@@ -0,0 +1,501 @@
+      String paramVal, String paramName,
+      List<NameValuePair> urlParameters) {
+    if (!StringUtils.isEmpty(paramVal)) {
+      StringBuilder sb = new StringBuilder(paramName);
+      sb.append(":");
+      sb.append(paramVal);
+      urlParameters.add(new BasicNameValuePair("fq", sb.toString()));
+    }
+  }
+
+  private HttpRequestWrapper prepareHttpRequest(
+      List<NameValuePair> urlParameters, InetSocketAddress solrAddress,
+      String uri) {
+    // add request parameter, form parameters
+    urlParameters.add(new BasicNameValuePair("q", "*:*"));
+    urlParameters.add(new BasicNameValuePair("wt", "json"));
+    urlParameters.add(new BasicNameValuePair("fl",
+        "access, agent, repo, resource, resType, event_count"));
+    urlParameters.add(new BasicNameValuePair("fq", "access:read"));
+    urlParameters.add(new BasicNameValuePair("fq", "repo:cm_ozone"));
+
+    urlParameters.add(new BasicNameValuePair("sort", "event_count desc"));
+    urlParameters.add(new BasicNameValuePair("start", "0"));
+    urlParameters.add(new BasicNameValuePair("rows", "0"));
+
+    urlParameters.add(new BasicNameValuePair("json.facet", "{\n" +
+        "    resources:{\n" +
+        "      type : terms,\n" +
+        "      field : resource,\n" +
+        "      sort : \"read_access_count desc\",\n" +
+        "      limit : 100,\n" +
+        "      facet:{\n" +
+        "        read_access_count : \"sum(event_count)\"\n" +
+        "      }\n" +
+        "    }\n" +
+        "  }"));
+    HttpRequestWrapper requestWrapper =
+        new HttpRequestWrapper(solrAddress.getHostName(),
+            solrAddress.getPort(), uri,
+            urlParameters, HttpRequestWrapper.HttpReqType.POST);
+    return requestWrapper;
+  }
+
+  private EntityReadAccessHeatMapResponse generateHeatMap(
+      EntityMetaData[] entities) {
+    EntityReadAccessHeatMapResponse rootEntity =
+        new EntityReadAccessHeatMapResponse();
+    rootEntity.setMinAccessCount(entities[0].getReadAccessCount());
+    rootEntity.setLabel("root");
+    List<EntityReadAccessHeatMapResponse> children =
+        rootEntity.getChildren();
+    Arrays.stream(entities).forEach(entityMetaData -> {
+      String path = entityMetaData.getVal();
+      String[] split = path.split("/");
+      if (split.length == 0) {
+        return;
+      }
+      long keySize = 0;
+      try {
+        keySize = getEntitySize(path);
+      } catch (IOException e) {
+        LOG.error("IOException while getting key size for key : " +
+            "{} - {}", path, e);
+      }
+      EntityReadAccessHeatMapResponse volumeEntity = null;
+      List<EntityReadAccessHeatMapResponse> volumeList =
+          children.stream().filter(entity -> entity.getLabel().
+              equalsIgnoreCase(split[0])).collect(Collectors.toList());
+      if (volumeList.size() > 0) {
+        volumeEntity = volumeList.get(0);
+      }
+      if (null != volumeEntity) {
+        addBucketData(rootEntity, volumeEntity, split,
```

Comment:
split array size requried to be "2", but no such check. Above check can have size to be "1" also for volume.

Reply:
> split array size requried to be "2", but no such check. Above check can have size to be "1" also for volume.

This check is not required here because this data comes directly from Audit logs from solr query and it will never be empty because audit logs are done for keys at leaf level always. However just for safety, size of 0 is already checked.